lib/fluent/plugin/out_s3.rb

```diff
@@ -630,6 +631,28 @@ def compress(chunk, tmp)
       end
     end
 
+    class ZstdCompressor < Compressor
+      def ext
+        'zst'.freeze
+      end
+
+      def content_type
+        'application/x-zst'.freeze
+      end
+
+      def compress(chunk, tmp)
+        uncompressed_data = ''
+        chunk.open do |io|
+          uncompressed_data = io.read
+        end
+        compressed_data = Zstd.compress(uncompressed_data, level: @level)
```

Comment:
> compressed_data = Zstd.compress(uncompressed_data, level: @level)

It appears that `@level` is always `nil`.

Reply:
@ddukbg What about this point?

Reply:
@daipom Oh, I forgot to mention.
I'll check that out and comment back.

Reply:
@daipom 

Hello :)

As per your feedback, I've fixed the issue where `@level` was always `nil`. I've added a `DEFAULT_LEVEL` constant with a value of 3 to the `ZstdCompressor` class and updated the `initialize` method to properly initialize the compression level.

- **Summary of changes:**
  - Added `DEFAULT_LEVEL` constant (value: 3)
  - Fixed compression level initialization
  - Handled line endings and encoding
  - Improved error handling

The commit message is:

```
fix: Add proper compression level handling to ZstdCompressor

- Add DEFAULT_LEVEL constant with value 3
- Fix compression level initialization
- Handle line endings and encoding
- Add proper error handling
```

Please review the updated code.

Thank you.